### PR TITLE
Hab suite config requires default

### PR DESCRIPTION
**Pull request by vincentpierre**
## Motivation and Context

Context:
The `habitat/config/default.py` config is important to the user since it contains all the defaults of the configurations loaded with `habitat.get_config`. As such, it is important that this file contains ALL possible keys the various tasks of habitat may use. This PR is meant to ensure that ALL keys that will be added to config files are present in the default (enforced in CI). Some config file can be manually excluded if they contain non-default keys on purpose.

Changes:
 - A new unit test has been added to make sure that any `.yaml` file in the `config` folder contains only keys that are present in the default config `habitat/config/default.py`.
 - In addition to the test, some keys of configs have been renamed for consistency across files.
 - Keys present in the configs that were not in the default have been added

TODO:
 - The current default values contain my best guesses, if a value needs to be changed, please leave a comment.
 - The default config file is organized based off my best effort. If there is a better way to organize the keys, please leave a comment.
 - The default config of the baselines was left unchanged in this PR and will be modified in a subsequent one.

## How Has This Been Tested

The change has been unit tested. 

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/test_config.py [hunk 1]
```diff
@@ -46,3 +51,33 @@ def test_overwrite_options():
...
+    "config_path",
+    glob("configs/**/*.yaml", recursive=True),
+)
+def test_no_core_config_has_non_default_keys(config_path):
+    configs_allowed_to_have_non_default_keys = [
+        # new_keys_test.yaml excluded since it explicitely uses
+        # keys not present in the default for testing purposes
+        "configs/test/new_keys_test.yaml",
+        # Planning Domain Definition Language configs are
+        # excluded since they do not implement the default config
```
Comment (mathfac) [inline test/test_config.py:66]: Maybe exclude pddl folder.

**Review by mathfac — APPROVED** [review]
This PR is just on point: adding default configs for Habitat 2.0 and adding the test to make sure in future PRs we aren't missing adding defaults. 
@ASzot do you have more granular comment here?

**Comment by mathfac** [comment]
We should make CI green before merge.

**Comment by vincentpierre** [comment]
Re-implemented with changes on #776

**Pull request closed without merging**